tests/integration/backend/filestate/backend_azure_test.go

```diff
@@ -0,0 +1,84 @@
+package filestate
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/pulumi/pulumi/pkg/v3/backend/filestate"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/testing/diagtest"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+//nolint:paralleltest // this test sets the global login state
+func TestAzureLoginSasToken(t *testing.T) {
+	cloudURL := "azblob://pulumitesting?storage_account=pulumitesting"
+	err := workspace.DeleteAccount(cloudURL)
+	assert.NoError(t, err)
+	assert.NotEmpty(t, os.Getenv("AZURE_STORAGE_SAS_TOKEN"),
+		"an azure storage SAS token needs to be set in the environment")
+
+	t.Cleanup(func() {
+		err := workspace.DeleteAccount(cloudURL)
+		assert.NoError(t, err)
+		fmt.Println("cleaned up 1")
+	})
+	ctx := context.Background()
+	project := workspace.Project{
+		Name:    "testing-sas-login",
+		Runtime: workspace.NewProjectRuntimeInfo("nodejs", nil),
+	}
+	fmt.Println(os.Environ())
+	backend, err := filestate.New(ctx, diagtest.LogSink(t), cloudURL, &project)
```

Comment:
Wonder if we should actually try to make a stack here to really test it?

Reply:
That sounds like a good idea.